lib/miq_automation_engine/engine/miq_ae_method_service.rb

```diff
@@ -2,7 +2,7 @@ module MiqAeMethodService
   def self.const_missing(name)
     super unless defined?(MiqAeServiceModelBase)
 
-    MiqAeServiceModelBase.create_service_model_from_name(name) || super
+    (defined?(MiqAeServiceModelBase) ? MiqAeServiceModelBase.create_service_model_from_name(name) : nil) || super
```

Comment:
Isn't the `defined?(MiqAeServiceModelBase)` already being done on line 3?

```ruby
super unless defined?(MiqAeServiceModelBase)
```

Basically, I don't see how this extra (redundant?) check is helping.  Am I missing something?

Reply:
ok. put as a guard clause. thanks

Reply:
No... that isn't the problem.  I am saying the guard clause seems to already exist, so what value are you adding with this PR?  **How is this PR helping?**

Reply:
> ok. put as a guard clause. thanks

Okay, this makes more sense now seeing the code.  Before we weren't doing a `return` with the above, so it was "calling" super, but was still continuing with the rest of the code.  Sorry, was not sure what you were getting at with the above, but is more clear seeing the code.

👍